### PR TITLE
implement Display for Partition, Segment and Stream

### LINKED ISSUE #481: impl `Display` for `Partition`, `Segment`, `Stream`
So that there will `no less` info than currently.

This is already done for topic:
```rust
impl fmt::Display for Topic {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> std::fmt::Result {
        write!(f, "ID: {}, ", self.topic_id)?;
        write!(f, "stream ID: {}, ", self.stream_id)?;
        write!(f, "name: {}, ", self.name)?;
        write!(f, "path: {}, ", self.path)?;
        write!(f, "partitions count: {:?}, ", self.partitions.len())?;
        write!(f, "message expiry (s): {:?}, ", self.message_expiry_secs)?;
        write!(f, "max topic size (B): {:?}, ", self.max_topic_size_bytes)?;
        write!(f, "replication factor: {}, ", self.replication_factor)
    }
}
```

and usage:
```rust
info!("Created topic {}", topic);
info!("Updated topic: {}", topic);
info!("Loading consumer groups for topic {} from disk...", topic);
info!("Loading topic {} from disk...", topic);
info!("Loaded topic {}", topic);
info!("Saved topic {}", topic);
info!("Deleting topic {}...", topic);
```

The aim of this task is to also use those calls whenever possible.

### PR DESCRIPTION
**Pull request by berendsliedrecht**
- Implemented `fmt::Display` for `Partition`, `Segment` and `Stream`
- Renamed `XXX_indexes` to `XXX_indices`
- Renamed `topics_ids` to `topic_ids`
- Closes #481

### CODE REVIEW

File: server/src/streaming/partitions/messages.rs [hunk 1]
```diff
@@ -41,11 +41,11 @@ impl Partition {
 
         let mut maybe_start_offset = None;
         for segment in self.segments.iter() {
-            if segment.time_indexes.is_none() {
+            if segment.time_indices.is_none() {
                 continue;
             }
 
-            let time_indexes = segment.time_indexes.as_ref().unwrap();
+            let time_indexes = segment.time_indices.as_ref().unwrap();
```
Comment (hubcio) [inline server/src/streaming/partitions/messages.rs]: let time_indices

**Review by hubcio — CHANGES_REQUESTED** [review]

**Comment by hubcio** [comment]
@berendsliedrecht please take a look at discord 🦀

**Comment by hubcio** [comment]
image

**Comment by hubcio** [comment]
hey @berendsliedrecht, any plans to continue?

**Comment by ynachi** [comment]
@hubcio, I would like to take this one if still relevant and available.

**Comment by hubcio** [comment]
> @hubcio, I would like to take this one if still relevant and available.

hey, yes, it's available. sorry for late reply, better join our discord to get instant replies :)

**Pull request closed without merging**